File: 프로그래머스/3/49191. 순위/순위.java
public class Solution {
    public static int solution(int n, int[][] results) {
        int answer = 0;
        int[][] graph = new int[n + 1][n + 1];
        
        for (int[] result : results) {
            int win = result[0];
            int lose = result[1];
            graph[win][lose] = 1;
            graph[lose][win] = -1;
        }
        
        for (int k = 1; k <= n; k++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    if (i == j || graph[i][j] != 0) {
                        continue;
                    }
                    if (graph[i][k] == 1 && graph[k][j] == 1) {
                        graph[i][j] = 1;
                        graph[j][i] = -1;
                    }
                }
            }
        }
        
        for (int i = 1; i <= n; i++) {
            int count = 0;
            for (int j = 1; j <= n; j++) {
                if (graph[i][j] == 0) {
                    count++;
                }
            }
            if (count == 1) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/3/49191. 순위/순위.py
def solution(n: int, results: list) -> int:
    answer = 0
    graph = [[0] * (n + 1) for _ in range(n + 1)]
    
    for win, lose in results:
        graph[win][lose] = 1
        graph[lose][win] = -1
    
    for k in range(1, n + 1):
        for i in range(1, n + 1):
            for j in range(1, n + 1):
                if i == j or graph[i][j] != 0:
                    continue
                if graph[i][k] == graph[k][j] == 1:
                    graph[i][j] = 1
                    graph[j][i] = -1
                    
    for row in graph:
        if row.count(0) == 2:
            answer += 1
    
    return answer